tss-esapi/src/interface_types/algorithm.rs

```diff
@@ -335,6 +336,28 @@ impl TryFrom<TPMI_ALG_ASYM> for AsymmetricAlgorithm {
     }
 }
 
+/// Enum representing the asymmetric algorithm interface type with specific properties.
+/// Use this instead of [AsymmetricAlgorithm], which will be deprecated at some point.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AsymmetricAlgorithmSelection {
+    Rsa(RsaKeyBits),
+    Ecc(EccCurve),
+}
+
+impl TryFrom<AsymmetricAlgorithm> for AsymmetricAlgorithmSelection {
```

Comment:
Could you add some public-facing documentation here that the conversion makes some assumptions about what RSA size/ECC curve you want?

Reply:
Done. I also removed the deprecation notice for `AsymmetricAlgorithm`, because it is just a direct mapping of the TSS esapi enum.